src/backend/routers/auth.py

```diff
@@ -317,3 +317,74 @@ def log_and_redirect_err(error_message: str):
     response = RedirectResponse(redirect_uri)
 
     return response
+
+
+@router.delete("/tool/auth/{tool_id}")
+async def delete_tool_auth(
+    tool_id: str,
+    request: Request,
+    session: DBSessionDep,
+    ctx: Context = Depends(get_context),
+):
+    """
+    Endpoint to delete Tool Authentication.
+
+    If completed, the corresponding ToolAuth for the requesting user is removed from the DB.
+
+    Args:
+        tool_id (str): ToolID to be deleted for the user. (eg. google_drive) Should be one of the values listed in the ToolName string enum class.
+        request (Request): current Request object.
+        session (DBSessionDep): Database session.
+        ctx (Context): Context object.
+
+    Returns:
+        {"status": "success"} on successful deletion.
+
+    Raises:
+        HTTPException: If there was an error deleting the tool auth.
+    """
+
+    logger = ctx.get_logger()
+
+    def log_and_return_error(error_message: str):
+        logger.error(event=error_message)
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"{error_message}",
+        )
+
+    user_id = ctx.get_user_id()
+
+    tool_id = tool_id.lower()
+
+    if user_id is None or user_id == "" or user_id == "default":
+        log_and_return_error("User ID not found.")
+
+    if tool_id not in [tool_name.value for tool_name in ToolName]:
+        log_and_return_error(
+            "tool_id must be present in the path of the request and must be a member of the ToolName string enum class."
+        )
+
+    tool = AVAILABLE_TOOLS.get(tool_id)
+
+    if tool is None:
+        log_and_return_error(f"Tool {tool_id} is not available in AVAILABLE_TOOLS.")
+
+    if tool.auth_implementation is None:
+        log_and_return_error(
+            f"Tool {tool.name} does not have an auth_implementation required for Tool Auth Deletion."
+        )
+
+    try:
+        tool_auth_service = tool.auth_implementation()
+        is_delete_tool_auth_successful = tool_auth_service.delete_tool_auth(
+            session, user_id
+        )
+
+        if not is_delete_tool_auth_successful:
+            log_and_return_error("Error deleting Tool Auth.")
+
+    except Exception as e:
+        log_and_return_error(str(e))
+
+    return {"status": "success"}
```

Comment:
Here as mentioned above, let's return the pydantic schema `DeleteToolAuth`, you can check other delete endpoints for examples. This should return just a `{}`, the status code of 200 will be explicit enough that it was successful.